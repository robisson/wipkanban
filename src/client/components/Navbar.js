import React from 'react';
import {Link} from 'react-router-dom'
import Logout from './Login/Logout'

const Navbar = () => {
    return (
        <nav
            className="navbar navbar-expand-lg navbar-toggleable-md navbar-light bg-faded">
            <button
                className="navbar-toggler navbar-toggler-right"
                type="button">
                <span className="navbar-toggler-icon"></span>
            </button>
            <a className="navbar-brand" href="#">WIPKanban</a>

            <div className="collapse navbar-collapse" id="navbarSupportedContent">
                <ul className="navbar-nav mr-auto">
                    <li className="nav-item active">
                        <a className="nav-link" href="#">Home</a>
                    </li>
                </ul>
                <div>
                    <Logout />
                </div>
            </div>
        </nav>

    )
}

export default Navbar;