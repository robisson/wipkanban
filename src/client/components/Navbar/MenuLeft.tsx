import * as React from "react";
import DeveloperBoardIcon from "@material-ui/icons/DeveloperBoard";
import Button from "@material-ui/core/Button";
import ArrowDropDownIcon from "@material-ui/icons/ArrowDropDown";
import { withStyles, StyleRulesCallback } from "@material-ui/core/styles";

const styles: StyleRulesCallback = () => ({
  navLeft: {
    marginLeft: 20
  }
});

type Props = {
  classes: any;
};

const MenuLeft: React.StatelessComponent<any> = ({ classes }: Props) => {
  return (
    <div className={classes.navLeft}>
      <Button
        color="inherit"
        size="small"
        style={{
          position: "relative",
          paddingTop: 15
        }}
      >
        <DeveloperBoardIcon />
        <small
          style={{
            position: "absolute",
            top: 0,
            left: 12,
            fontSize: 8
          }}
        >
          {"Guarida/Tecnologia da informação"}
        </small>
        Minhas atividades
        <ArrowDropDownIcon />
      </Button>
    </div>
  );
};

export default withStyles(styles)(MenuLeft);
