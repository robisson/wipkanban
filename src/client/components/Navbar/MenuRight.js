//@flow
import * as React from "react";
import { withStyles } from "@material-ui/core/styles";

const styles = {
  navRight: {
    position: "absolute",
    right: 5
  }
};

type Props = {
  classes: Object,
  children?: React.Node
};

const MenuRight = ({ classes, children }: Props) => {
  return <div className={classes.navRight}>{children}</div>;
};

export default withStyles(styles)(MenuRight);
