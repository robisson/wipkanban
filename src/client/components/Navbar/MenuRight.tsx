import * as React from "react";
import { withStyles } from "@material-ui/core/styles";

const styles: any = () => {
  return {
    navRight: {
      position: "absolute",
      right: 5
    }
  };
};

interface IMenuRightProps {
  classes: any;
  children?: any;
}

const MenuRight: React.StatelessComponent<any> = ({
  classes,
  children
}: IMenuRightProps) => {
  return <div className={classes.navRight}>{children}</div>;
};

export default withStyles(styles)(MenuRight);
