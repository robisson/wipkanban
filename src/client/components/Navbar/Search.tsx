import * as React from "react";
import SearchIcon from '@material-ui/icons/Search';
import IconButton from '@material-ui/core/IconButton';
import Tooltip from '@material-ui/core/Tooltip';

const Search = () => {
  return (
    <Tooltip title="Search cards">
      <IconButton color="inherit" aria-label="Search cards">
        <SearchIcon />
      </IconButton>
    </Tooltip>
  );
};

export default Search;
