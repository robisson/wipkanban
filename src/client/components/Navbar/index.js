// @flow
import * as React from "react";
import {withStyles} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';

const styles = {
  root: {
    flexGrow: 1
  },
  flex: {
    flex: 1
  },
  gridContainer: {
    padding: 2
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  }
};

type Props = {
  classes: Object,
  children?:React.Node
};

function Navbar(props : Props) {
  const {classes,children} = props;
  return (
    <AppBar position="fixed">
      <Toolbar>
        <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
          <MenuIcon/>
        </IconButton>
        <Typography variant="title" color="inherit">
          WIPLean
        </Typography>
        {children}

      </Toolbar>
    </AppBar>
  );
}

export default withStyles(styles)(Navbar);
