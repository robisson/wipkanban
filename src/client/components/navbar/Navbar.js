import React from 'react';
import {Link} from 'react-router-dom'
import UserNavbar from './UserNavbar';
import BtnAddResources from './BtnAddResources'
const Navbar = ({user}) => {

    return (
        <nav className="navbar navbar-expand-lg navbar-dark bg-murrey-dark py-1">
            <a className="navbar-brand" href="#">WIPKanban</a>
            <button className="navbar-toggler navbar-toggler-right" type="button">
                <span className="navbar-toggler-icon"></span>
            </button>
            <div className="collapse navbar-collapse">
                <ul className="navbar-nav mr-auto">
                    <li className="nav-item active">
                        <a href="#" className="nav-link">
                            Home
                        </a>
                    </li>
                    <li className="nav-item">
                        <a href="#" className="nav-link">
                            Companies
                        </a>
                    </li>
                    <li className="nav-item">
                        <a href="#" className="nav-link">
                            Projects
                        </a>
                    </li>
                    <li className="nav-item">
                        <a href="#" className="nav-link">
                            Boards
                        </a>
                    </li>
                    <li className="nav-item">
                        <a href="#" className="nav-link">
                            Teams
                        </a>
                    </li>
                </ul>
                <div>
                    <ul className="navbar-nav mr-auto">
                        <BtnAddResources/> {user && <UserNavbar user={user}/>}
                    </ul>

                </div>
            </div>
        </nav>

    )
}

export default Navbar;