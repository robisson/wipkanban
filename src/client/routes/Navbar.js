// @flow
import * as React from "react";
import {withStyles} from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import IconButton from '@material-ui/core/IconButton';
import MenuIcon from '@material-ui/icons/Menu';
import SearchIcon from '@material-ui/icons/Search';
import Avatar from '@material-ui/core/Avatar';

const styles = {
  root: {
    flexGrow: 1
  },
  flex: {
    flex: 1
  },
  gridContainer: {
    padding: 2
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20
  }
};

type Props = {
  classes: Object
};

function Navbar(props : Props) {
  const {classes} = props;
  return (
    <AppBar position="static">
      <Toolbar>
        <IconButton className={classes.menuButton} color="inherit" aria-label="Menu">
          <MenuIcon/>
        </IconButton>
        <Typography variant="title" color="inherit" className={classes.flex}>
          WipKanban
        </Typography>
        <IconButton color="inherit" aria-label="Menu">
          <SearchIcon/>
        </IconButton>
        <IconButton color="inherit" aria-label="Menu">
          <Avatar
            aria-label="Robisson Oliveira"
            src="/uploads/7b435896-aeca-4660-aaa3-b734e3e9200a.jpg"/>
        </IconButton>
      </Toolbar>
    </AppBar>
  );
}

export default withStyles(styles)(Navbar);
