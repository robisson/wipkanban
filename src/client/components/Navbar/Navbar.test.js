import React from "react";
import Navbar from "./index";
import { createShallow } from "@material-ui/core/test-utils";
import AppBar from '@material-ui/core/AppBar';

// setup file
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Navbar Component", () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({dive:true});
  });

  it("should render a AppBar Component", () => {
    let wrapper = shallow(<Navbar />);
    expect(wrapper.find(AppBar).type()).toEqual(AppBar);
  });

  it("should render its children", () => {
    let wrapper = shallow(<Navbar><div>Hello</div></Navbar>);
    expect(wrapper.find('div').type()).toEqual('div');
    expect(wrapper.find('div').at(0).text()).toEqual('Hello');
  });

});
