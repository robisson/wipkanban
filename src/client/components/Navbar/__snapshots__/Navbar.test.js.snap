// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should render a AppBar Component 1`] = `
<WithStyles(AppBar)
  position="fixed"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      aria-label="Menu"
      className="Navbar-menuButton-4"
      color="inherit"
    >
      <pure(MenuIcon) />
    </WithStyles(IconButton)>
    <WithStyles(Typography)
      color="inherit"
      variant="title"
    >
      WIPLean
    </WithStyles(Typography)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;

exports[`Navbar Component should render its children 1`] = `
<WithStyles(AppBar)
  position="fixed"
>
  <WithStyles(Toolbar)>
    <WithStyles(IconButton)
      aria-label="Menu"
      className="Navbar-menuButton-4"
      color="inherit"
    >
      <pure(MenuIcon) />
    </WithStyles(IconButton)>
    <WithStyles(Typography)
      color="inherit"
      variant="title"
    >
      WIPLean
    </WithStyles(Typography)>
    <div>
      Hello
    </div>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
